### ISSUE TITLE
nillable=true causes the data type to be misread

### ISSUE THREAD

**Issue by matthewsac** [fsprojects/FSharp.Data.Xsd#25]
### Description

In the following XSD snippet in partiesFile.xsd:

    <xs:complexType name="passportType">
        <xs:sequence>
            <xs:element nillable="true" name="PassportCountry" type="xs:string"/>
            <xs:element nillable="true" name="PassportNumber" type="xs:string"/>
            <xs:element nillable="true" name="PassportExpirationDate" type="xs:string"/>
        </xs:sequence>
    </xs:complexType>

I read partiesFile in with the following code:

type partiesSchema = XmlProvider<Schema = PartiesFile>
type passport = partiesSchema.Passport

Hover over Passport in VS2017 and it says:

type Passport =
inherit XmlElement
new: passPortCountry: PassportCountry * passportNumber: PassportNumber * passportExpirationDate: PassportExpirationDate -> Passport + 1 overload
member PassportCountry: PassportCountry
member PassportExpirationDate: PassportExpirationDate
member PassportNumber : Passport Number

In short, it's reading the name attribute as the type and ignoring the type="xs:string".

These were the only ones in the schema that had nillable="true" as attributes. When I removed these, it correctly recognized them as string types.

Position of the attribute doesn't seem to matter. I tried:
<xs:element name="PassportCountry" type="xs:string" nillable="true" />

This still didn't work.

### Expected behavior

Declared types are recognized despite the presence of the nillable attribute

* Windows10
* master
* .NET 4.5.2, Fsharp.Core 4.4.0.0

**Comment by giacomociti** [comment]
Nillable elements are strange, for example the following is valid according to your schema:

    <Passport>
      <PassportCountry>XY</PassportCountry>
      <PassportNumber>123ABC</PassportNumber>
      <PassportExpirationDate xsi:nil="true"
            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"/>
    </Passport>

As you can see for the expiration date, the absence of a value must be specified explicitly with this strange `nil` attribute defined by the W3C. The type provider handles this creating a composite structure including `Nil` and  `Value`, both optional. So, for example, if `x` is the result of parsing the above xml, the following conditions hold:

```
x.PassportCountry.Nil = None
x.PassportCountry.Value = Some "XY"
x.PassportNumber.Nil = None
x.PassportNumber.Value = Some "123ABC"
x.PassportExpirationDate.Nil = Some true
x.PassportExpirationDate.Value = None

```

while this may seem weird (but I think the weirdness is first of all in the xsd specs), the type provider behaves this way to be consistent with the case when a schema is not used and the type is derived from samples.

**Comment by matthewsac** [comment]
Thanks for your attentiveness. You may close this issue.
